.github/workflows/Actions-workflow.yml

```diff
@@ -0,0 +1,229 @@
+            for file in sortedDictionary:
+              uniqFileName = workflowDetails[file]['name']
+              uniqFileName = uniqFileName.replace(' ','_')      #replace spaces with underscore
+              uniqFileName = uniqFileName.replace('(','_')      #replace parenthesis with underscore
+              uniqFileName = uniqFileName.replace(')','_')
+              uniqFileName = uniqFileName.upper()
+              fileBlock = f"{uniqFileName}[\"{workflowDetails[file]['name']}[{file}]\"]"
+              jobs = workflowDetails[file]['jobs']
+              markdownFile.write('```mermaid\n')
+              markdownFile.write('\nflowchart LR\n')
+              for job in jobs:
+                needs = jobs[job]['needs']
+                needs = needs.replace('[','')
+                needs = needs.replace(']','')
+                needs = needs.replace(' ','')
+                if len(needs) == 0:
+                  markdownFile.write(f"\n  {fileBlock} ---> {job}(\"{jobs[job]['name']}\")")
+                else:
+                  needs = list(needs.split(','))
+                  for need in needs:
+                    markdownFile.write(f"\n  {need}(\"{jobs[need]['name']}\") ---> {job}(\"{jobs[job]['name']}\")")
+                uses = jobs[job]['uses']  
+                if uses != "":
+                  uses = uses.split('/')[-1]
+                  str = ","
+                  inputs = str.join(jobs[job]['inputs'])
+                  inputs = inputs.replace('\"','')    #remove " (double quotes)
+                  inputs = inputs.replace("\'",'')    #remove ' (single quotes)
+                  if len(inputs) != 0:
+                    markdownFile.write(f"\n  {job}(\"{jobs[job]['name']}\ninputs:[{inputs}]\") ---> {uses}(\"{workflowDetails[uses]['name']}[{uses}]\")")
+                  else:
+                    markdownFile.write(f"\n  {job}(\"{jobs[job]['name']}\") ---> {uses}")
+              markdownFile.write('\n\n```\n\n')
+        
+            markdownFile.write('## GitHub Actions Trigger Overview\n')
+            triggersList = set()
+            for file in workflowDetails:
+              for trigger in workflowDetails[file]['trigger']:
+                triggersList.add(trigger)  #triggersList contains unique list of all types of triggers used in the workflow file
+                    
+            for trigger in sorted(triggersList):
+              markdownFile.write('\n```mermaid\n')
+              markdownFile.write('\nflowchart TB\n')
+              markdownFile.write(f'\n  subgraph {trigger.upper()}')
+              for file in sortedDictionary:
+                if trigger in workflowDetails[file]['trigger']:
+                  markdownFile.write(f'\n    {trigger} ---> {file}[\"{workflowDetails[file]["name"]}[{file}]\"]')
+              markdownFile.write('\n  end\n')
+              markdownFile.write('\n```\n')
+        shell: python
```

Comment:
Just for cleanliness, can we move this above the run: without breaking anything?  Other than that, everything looks great.

Reply:
Yes, it is possible to move it above run: